pkg/channel/consolidated/utils/util.go

```diff
@@ -115,7 +114,7 @@ func GetKafkaAuthData(ctx context.Context, secretname string, secretNS string) *
 	parseTls(secret, kafkaAuthConfig)
 	parseSasl(secret, kafkaAuthConfig)
 
-	return kafkaAuthConfig
+	return kafkaAuthConfig, nil
 }
```

Comment:
This seems odd - we're adding an `error` return value but then ALWAYS returning nil ?  What value does it provide?

Reply:
It's allowing the GetKafkaAuthData function to satisfy the sarama.GetAuth prototype so it can be passed in directly to the LoadSettings call in GetKafkaConfig().  Otherwise it would need a one-line wrapper to return the nil error (which I would be happy to do if that would look better).